app/src/main/java/com/moonpig/mvisample/bindingadapters/BindingAdapters.kt

```diff
@@ -0,0 +1,29 @@
+package com.moonpig.mvisample.bindingadapters
+
+import android.databinding.BindingAdapter
+import android.view.View
+import android.widget.ImageView
+import com.moonpig.mvisample.mvibase.Visibility
+import com.squareup.picasso.Picasso
+
+@BindingAdapter("imageUrl")
+fun loadImage(view: ImageView, url: String?) {
+    if (url == null || url.isEmpty())
+        return
+
+    Picasso.get().load(url).into(view)
+}
+
+@BindingAdapter("mappedVisibility")
+fun mapVisibility(view: View, visibility: Visibility?) {
```

Comment:
Can be simplified to something like this:
```kotlin
fun mapVisibility(view: View, visibility: Visibility?) {
    visibility?.toAndroidVisibility()?.let {
        view.visibility = it
    }
}
```
You don't really need to diff check it as it's the first thing View does when you call `setVisibility`.